packages/ui/src/machines/authenticator/index.ts

```diff
@@ -64,50 +69,92 @@ export function createAuthenticatorMachine() {
           ],
         },
         signIn: {
-          entry: 'spawnSignInActor',
-          exit: stopActor('signInActor'),
+          initial: 'spawnActor',
+          states: {
+            spawnActor: {
+              always: { actions: 'spawnSignInActor', target: 'runActor' },
+            },
+            runActor: {
+              entry: 'clearActorDoneData',
+              exit: stopActor('signInActor'),
+            },
+          },
           on: {
             SIGN_UP: 'signUp',
             RESET_PASSWORD: 'resetPassword',
             'done.invoke.signInActor': [
               {
                 target: 'signUp',
+                actions: 'setActorDoneData',
```

Comment:
After actor is done (line 85), immediately set any returned data to `ActorDoneData`